libs/platform/src/lib/components/dynamic-page/dynamic-page.component.ts

```diff
@@ -0,0 +1,348 @@
+    tabs: DynamicPageContentComponent[] = [];
+
+    /** @hidden */
+    private _subscriptions: Subscription = new Subscription();
+
+    /**
+     * @hidden
+     * subscription for when toggle header is called
+     */
+    private _toggleSubscription: Subscription;
+
+    /**
+     * @hidden
+     * subscription for when content is scrolled
+     */
+    private _scrollSubscription: Subscription;
+
+    /** @hidden */
+    constructor(
+        protected _cd: ChangeDetectorRef,
+        private _elementRef: ElementRef<HTMLElement>,
+        private _renderer: Renderer2,
+        private _dynamicPageService: DynamicPageService,
+        protected _dynamicPageConfig: DynamicPageConfig,
+        private scrollDispatcher: ScrollDispatcher,
+        private zone: NgZone
+    ) {
+        super(_cd);
+        this._toggleSubscription = this._dynamicPageService.$toggle.subscribe((val) => {
+            this.toggledVal = val;
+        });
+    }
+
+    /** @hidden */
+    ngOnInit(): void {
+        this._addClassNameToHostElement(CLASS_NAME.dynamicPage);
+    }
+
+    /** @hidden */
+    ngAfterContentInit(): void {
+        this._listenToChildrenQueryListChanges();
+        if (this.background) {
+            this.titleComponent.background = this.background;
+            this.headerComponent.background = this.background;
+            this.childcontent.background = this.background;
+        }
+        if (this.size) {
+            this.titleComponent.size = this.size;
+            this.headerComponent.size = this.size;
+            this.childcontent.size = this.size;
+        }
+    }
+
+    /**@hidden */
+    ngAfterViewInit(): void {
+        this._setTabStyles();
+        this._setToolbarStyles();
+        this._listenToChildrenQueryListChanges();
+
+        this._subscriptions.add(
+            this.tabbedContent.changes.subscribe(() => {
+                this._listenToChildrenQueryListChanges();
+                this._setTabStyles();
+            })
+        );
+        if (this.headerComponent?.collapsible) {
+            this.snapOnScroll();
+        }
+        this._cd.detectChanges();
+    }
+
+    /**@hidden */
+    ngOnDestroy(): void {
+        this.scrollDispatcher.deregister(this.scrollable);
+        this._toggleSubscription.unsubscribe();
+        this._scrollSubscription.unsubscribe();
+        this._subscriptions.unsubscribe();
+    }
+
+    /**
+     * Snap the header to expand or collapse based on scrolling. Uses CDKScrollable.
+     */
+    snapOnScroll(): void {
+        if (!this.scrollDispatcher.scrollContainers.has(this.scrollable)) {
+            this.scrollDispatcher.register(this.scrollable);
+            const scrollContainers = this.scrollDispatcher.getAncestorScrollContainers(this.scrollable.getElementRef());
+            scrollContainers.forEach((element) => {
+                if (element !== this.scrollable) {
+                    this.scrollDispatcher.deregister(element);
+                }
+            });
```

Comment:
there is a global level `cdkScrollable` applied to the documentation in class `platform-documentation.component.html` which calls our scroll logic on even page scroll. To not listen for this page-level cdkScrollable, I am trying to deregister any cdkScrollable that is not related to Dynamic Page.

The way cdkscrollable is used might be flawed since I couldn't find much good documentation on how to use cdkScrollable without angular material context.